src/interface_adapter/display_profile/DisplayProfileController.java

```diff
@@ -0,0 +1,17 @@
+package interface_adapter.display_profile;
+
+import use_case.display_profile.DisplayProfileInputBoundary;
+import use_case.display_profile.DisplayProfileInputData;
+
+public class DisplayProfileController {
+    final DisplayProfileInputBoundary displayProfileUseCaseInteractor;
+
+    public DisplayProfileController(DisplayProfileInputBoundary displayProfileUseCaseInteractor) {
+        this.displayProfileUseCaseInteractor = displayProfileUseCaseInteractor;
+    }
+
+    public void execute(String username, String otherUsername) {
```

Comment:
We only put the info from user input into the input data object, so the username (our current user) should be passed to the interactor directly (not in input data object).